File: driving-school-web/src/Types/student.ts
export type Student = {
    id?: number;
    name: string;
    lastName: string;
    yearOfBirth: number;
    email: string;
    phoneNumber: string;
    address: string;
    city: string;
    trainingCategory: string;
    dateOfRegistration: Date;
    dateOfTheoryExam?: Date;
    dateOfDrivingExam?: Date;
    theoryMark?: number;
    drivingMark?: number;
}

File: DrivingSchoolApi/DrivingSchool.Core/Models/Student.cs

namespace DrivingSchool.Core.Models
{
    public class Student: Entity
    {
        public string Name { get; set; }
        public string LastName { get; set; }
        public int YearOfBirth { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public DateTime DateOfRegistration { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string  TrainingCategory { get; set; }
        public DateTime? DateOfTheoryExam { get; set; }
        public DateTime? DateOfDrivingExam { get; set; }
        public int? TheoryMark { get; set; }
        public int? DrivingMark { get; set; }
        public string? UniqueTheoryCode { get; set; }
        public string? UniqueDrivingCode { get; set; }
    }
}